src/main/java/edu/cornell/kfs/fp/batch/service/impl/AccountingXmlDocumentDownloadAttachmentServiceImpl.java

```diff
@@ -69,18 +70,24 @@ public Attachment createAttachmentFromBackupLink(GloballyUnique parentObject, Ac
                 throw new ValidationException(UNABLE_TO_DOWNLOAD_ATTACHMENT_MESSAGE + accountingXmlDocumentBackupLink.getLinkUrl());
             }
         } catch (IOException e) {
-            LOG.error("createAttachmentFromBackupLink, Unable to download attachment: " + accountingXmlDocumentBackupLink.getLinkUrl(), e);
-            throw new ValidationException(UNABLE_TO_DOWNLOAD_ATTACHMENT_MESSAGE + accountingXmlDocumentBackupLink.getLinkUrl());
+            return handleFileException(e, accountingXmlDocumentBackupLink);
         } catch (IllegalArgumentException iae) {
             if (StringUtils.equalsIgnoreCase(CUKFSConstants.ANTIVIRUS_FAILED_MESSAGE, iae.getMessage())) {
                 throw new ValidationException("Unable to download attachment due to failing antivirus scan: " + accountingXmlDocumentBackupLink.getLinkUrl());
             } else {
                 LOG.error("createAttachmentFromBackupLink, Unable to download attachment due to illegal argument exception: " + accountingXmlDocumentBackupLink.getLinkUrl(), iae);
                 throw new ValidationException(UNABLE_TO_DOWNLOAD_ATTACHMENT_MESSAGE + accountingXmlDocumentBackupLink.getLinkUrl());
             }
+        } catch (ProcessingException processingException) {
+            return handleFileException(processingException, accountingXmlDocumentBackupLink);
         }
     }
 
+    protected Attachment handleFileException(Exception e, AccountingXmlDocumentBackupLink backupDocLink) throws ValidationException {
+        LOG.error("createAttachmentFromBackupLink, Unable to download attachment: " + backupDocLink.getLinkUrl(), e);
+        throw new ValidationException(UNABLE_TO_DOWNLOAD_ATTACHMENT_MESSAGE + backupDocLink.getLinkUrl());
```

Comment:
I think we need to adjust this some more.  One of the goals of this Jira is to help the person receiving this report know that they need to contact the IT people who support their system so they can fix the SSL certificate error, or what whatever might be wrong.

Reply:
Please see my latest commit